### PR TITLE
CORTX-28862:conf directory changes

### PR DESCRIPTION
**Pull request by kailashyadav11**
Signed-off-by: Kailash Yadav <kailash.yadav@seagate.com>

# Problem Statement
- Changes needed for maintaining a better structure for gconf

# Design
-  https://seagate-systems.atlassian.net/wiki/spaces/PUB/pages/651132980/CORTX+GConf+Config+Store+Keys

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [x] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Comment by cortx-admin** [comment]
Can one of the admins verify this patch?

File: conf/gconf/cluster.conf.sample [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+        MG12NU5BVjBVeWpBSWJuSV9uVjV5UzUxN3lTSVNkNUk1YkRzbjd2QkE9PQ==
+    kafka:
+      admin: admin
+      num_endpoints: 2
+      endpoints:
+      - kafka-server1.cortx-cluster.com
+      - kafka-server2.cortx-cluster.com
+      secret: !!binary |
+        TmRPNThMSm5GOVl5RHgwTVFfaTRJZ3k3YUUwMldmNkdxRE1KYVIxc0E9PQ==
+    openldap:                                                 # deleted
```
Comment (tanujashinde0405) [inline conf/gconf/cluster.conf.sample:21]: Mark keys as #deprecated instead of #deleted

**Review by tanujashinde0405 — CHANGES_REQUESTED** [review]

File: conf/gconf/cluster.conf.sample [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+    hostname: server-node1
+    name: server-node1
+    cluster_id: 0007ec45379e36d9fa089a3d615c32a3
+    num_components: 3
+    components:
+    - name: utils
+      version: 2.0.0-33
+    - name: hare
+      version: 2.0.0-47
+    - name: s3
```
Comment (tanujashinde0405) [inline conf/gconf/cluster.conf.sample:288]: add s3 services section here as it is there in leagy_s3 deployment

**Review by tanujashinde0405 — CHANGES_REQUESTED** [review]

**Review by ujjwalpl — APPROVED** [review]

**Review by nitin-seagate — CHANGES_REQUESTED** [review]
Mention old keys as "deprectaed"

File: conf/gconf/cluster.conf.sample (hunk 1 above)
Comment (kailashyadav11) [reply conf/gconf/cluster.conf.sample]: > Mention old keys as "deprecated"

done

File: conf/gconf/cluster.conf.sample (hunk 2 above)
Comment (kailashyadav11) [reply conf/gconf/cluster.conf.sample:295]: done

**Review by tanujashinde0405 — COMMENTED** [review]
Chnages will be needed in config.yaml.smaple file also, for new sections added. 
i.e ha section as well as limits section for all components, hare->hax->endpoints.

**Review by tanujashinde0405 — APPROVED** [review]

**Review by nitin-seagate — APPROVED** [review]
LGTM

**Pull request merged**